src/common/sensor.rs

```diff
@@ -0,0 +1,75 @@
+#![allow(dead_code)]
+
+use crate::common::status::Status;
+use std::collections::BTreeMap;
+use std::collections::HashMap;
+
+pub type GenericReadingsResult =
+    ::std::collections::HashMap<::prost::alloc::string::String, ::prost_types::Value>;
+
+pub type TypedReadingsResult<T> = ::std::collections::HashMap<String, T>;
+
+pub trait Sensor: Status {
+    fn get_generic_readings(&self) -> anyhow::Result<GenericReadingsResult>;
+}
+
+pub trait SensorT<T>: Sensor {
+    fn get_readings(&self) -> anyhow::Result<TypedReadingsResult<T>>;
+}
+
+// A local wrapper type we can use to specialize `From` for `prost_types::Value``
+pub struct SensorResult<T> {
+    pub value: T,
+}
+
+impl From<SensorResult<f64>> for ::prost_types::Value {
```

Comment:
I could add some more definitions here for other builtins if we think that is useful.